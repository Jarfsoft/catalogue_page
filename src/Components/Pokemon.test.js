import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom'; import Pokemon from './Pokemon';

describe('Component render', () => {
  const initialState = {
    list: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12],
    type: '',
  };
  const mockStore = configureStore();
  let store;

  it('Pokemon component is rendered', () => {
    store = mockStore(initialState);
    const pokemon = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <Pokemon id={1} />
          </Provider>
        </BrowserRouter>,
      ).toJSON();
    expect(pokemon).toMatchSnapshot();
  });

  it('should containt an image', () => {
    store = mockStore(initialState);
    const pokemon = render(
      <BrowserRouter>
        <Provider store={store}>
          <Pokemon id={1} />
        </Provider>
      </BrowserRouter>,
    );
    const img = pokemon.container.querySelectorAll('img');
    expect(img).toHaveLength(1);
  });
});
