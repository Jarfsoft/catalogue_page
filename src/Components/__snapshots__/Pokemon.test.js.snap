// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component render Pokemon component is rendered 1`] = `
<div
  className="poke-card"
>
  <h1>
    
  </h1>
  <img
    alt="Pokemon sprite"
    src=""
  />
  <a
    href="/pokemon/1"
    onClick={[Function]}
  >
    Details
  </a>
</div>
`;
